website/src/controllers/groups.js

```diff
@@ -136,8 +136,29 @@ api.get = function(req, res, next) {
   populateQuery(gid, q);
   q.exec(function(err, group){
     if (err) return next(err);
-    if (!group && gid!=='party') return res.json(404,{err: "Group not found or you don't have access."});
-    res.json(group);
+    if (!group && group != null && gid!=='party') return res.json(404,{err: "Group not found or you don't have access."});
+
+    //Since we have a limit on how many members are populate to the group, we want to make sure the user is always in the group
+    var userInGroup = _.find(group.members, function(member){ return member._id == user._id; });
+
+    //If the group is private or the group is a party, then the user must be a member of the group based on access restrictions above
+    if (group.privacy === 'private' && gid !== 'party') {
```

Comment:
Should we just remove the `gid !== party? ` ? I think this line was trying to say, if you access the party page, you must have access to the party, but the query above prevents you from seeing the group if you don't have access.